components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/par/OAuth2ParEndpoint.java

```diff
@@ -144,6 +154,38 @@ private void handleValidation(HttpServletRequest request, MultivaluedMap<String,
 
     private boolean isRequestUriProvided(MultivaluedMap<String, String> params) {
 
-        return params.containsKey(OAuthConstants.OAuth20Params.REQUEST_URI);
+        return params.containsKey(OAuthConstants.OAuth20Params.REQUEST_URI) &&
+                !params.get(OAuthConstants.OAuth20Params.REQUEST_URI).isEmpty();
```

Comment:
Is this check to see if a request_uri is also sent in the par request? in that case IMO we don't need to check if its empty or not. If the param key is there we can send the error back.